libraries-5/src/main/java/com/baeldung/yauaa/HomePageController.java

```diff
@@ -0,0 +1,37 @@
+package com.baeldung.yauaa;
+
+import java.util.List;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.servlet.ModelAndView;
+
+import nl.basjes.parse.useragent.UserAgent;
+import nl.basjes.parse.useragent.UserAgentAnalyzer;
+
+@Controller
+public class HomePageController {
+
+    private static final List<String> SUPPORTED_DEVICES = List.of("Mobile", "Tablet", "Phone");
+    private final UserAgentAnalyzer userAgentAnalyzer;
+
+    public HomePageController(UserAgentAnalyzer userAgentAnalyzer) {
+        this.userAgentAnalyzer = userAgentAnalyzer;
+    }
+
+    @GetMapping("/home")
```

Comment:
Since we reject non mobile devices, maybe putting mobile in the URL would make this clearer?

Or happy to be convinced otherwise that this is truly a mobile-only website example, just not something I've seen IRL as far as I remember.

```suggestion
    @GetMapping("/mobile/home")
```

Reply:
review incorporated.